src/include/optimizer/count_min_sketch.h

```diff
@@ -0,0 +1,124 @@
+
+#pragma once
+
+#include "common/logger.h"
+
+#include <murmur3/MurmurHash3.h>
+
+#include <cmath>
+#include <vector>
+#include <cassert>
+#include <cstdlib>
+#include <cinttypes>
+
+namespace peloton{
+namespace optimizer{
+
+class CountMinSketch {
+ public:
+
+  using SketchElemType = uint64_t;
+
+  const int depth;
+  const int width;
+  const double eps;
+  const double confidence;
+  uint64_t size;
```

Comment:
size is for counting the number of rows. I think we can get this information later from catalog.